src/structures/CommandInteraction.js

```diff
@@ -0,0 +1,155 @@
+'use strict';
+
+const APIMessage = require('./APIMessage');
+const Interaction = require('./Interaction');
+const Channel = require('../structures/Channel');
+const Role = require('../structures/Role');
+const User = require('../structures/User');
+const { ApplicationCommandOptionType } = require('../util/Constants');
+const Snowflake = require('../util/Snowflake');
+
+/**
+ * Represents a command interaction, see {@link InteractionClient}.
+ * @extends {Interaction}
+ */
+class CommandInteraction extends Interaction {
+  constructor(client, data, syncHandle) {
+    super(client, data);
+
+    this.syncHandle = syncHandle;
+
+    /**
+     * The ID of the invoked command.
+     * @type {Snowflake}
+     * @readonly
+     */
+    this.commandID = data.data.id;
+
+    /**
+     * The name of the invoked command.
+     * @type {string}
+     * @readonly
+     */
+    this.commandName = data.data.name;
+
+    const map = o => {
+      switch (o.type) {
+        case ApplicationCommandOptionType.SUB_COMMAND:
+          return {
+            name: o.name,
+            type: ApplicationCommandOptionType[o.type],
+            options: o.options.map(map),
+          };
+        case ApplicationCommandOptionType.SUB_COMMAND_GROUP:
+          return {
+            name: o.name,
+            type: ApplicationCommandOptionType[o.type],
+            options: o.options.map(map),
+          };
+        case ApplicationCommandOptionType.STRING:
+          return {
+            name: o.name,
+            type: ApplicationCommandOptionType[o.type],
+            value: o.value,
+          };
+        case ApplicationCommandOptionType.INTEGER:
+          return {
+            name: o.name,
+            type: ApplicationCommandOptionType[o.type],
+            value: o.value,
+          };
+        case ApplicationCommandOptionType.BOOLEAN:
+          return {
+            name: o.name,
+            type: ApplicationCommandOptionType[o.type],
+            value: o.value,
+          };
```

Comment:
since all these cases are the same you can combine them into one

```js
switch(x) {
  case 1:
  case 2:
  case 3:
    console.log("x is 1, 2, or 3");
}
```